front/components/assistant_builder/AssistantBuilderTablesModal.tsx

```diff
@@ -55,45 +114,199 @@ export default function AssistantBuilderTablesModal({
+      !selectedDataSource.connectorId ||
+      isInitializingNodes ||
+      !!selectedManagedTables.length ||
+      !Object.keys(tablesQueryConfiguration).length ||
+      !!nodeIntializationError
+    ) {
+      return;
+    }
+    const tableConfigs = Object.values(tablesQueryConfiguration).filter(
+      (c) => c.dataSourceId === selectedDataSource.name
+    );
+    if (!tableConfigs.length) {
+      return;
+    }
+    setIsInitializingNodes(true);
+    void initializeManagedTables(selectedDataSource, tableConfigs)
+      .catch((e) => {
+        setNodeIntializationError(true);
+        throw e;
+      })
+      .finally(() => {
+        setIsInitializingNodes(false);
+      });
+  }, [
+    selectedDataSource,
+    tablesQueryConfiguration,
+    isInitializingNodes,
+    selectedManagedTables.length,
+    initializeManagedTables,
+    nodeIntializationError,
+  ]);
+
+  useEffect(() => {
+    if (
+      !selectedDataSource ||
+      !selectedDataSource.connectorId ||
+      isInitializingParents ||
+      !!parentsIntializationError ||
+      !!Object.keys(parentsById).length ||
+      !selectedManagedTables.length
+    ) {
+      return;
+    }
+    setIsInitializingParents(true);
+    void initializeParents(
+      selectedManagedTables.map((n) => n.internalId),
+      selectedDataSource
+    )
+      .catch((e) => {
+        setParentsIntializationError(true);
+        throw e;
+      })
+      .finally(() => {
+        setIsInitializingParents(false);
+      });
+  }, [
+    selectedDataSource,
+    selectedManagedTables,
+    parentsById,
+    initializeParents,
+    isInitializingParents,
+    parentsIntializationError,
+  ]);
+
+  async function save() {
+    if (!selectedDataSource) {
+      return;
+    }
+    setIsSaving(true);
+
+    try {
+      const tableIds = removeNulls(
+        selectedManagedTables.map((n) => n.dustTableId)
+      );
+      const tables = await Promise.all(
+        tableIds.map(async (id) => {
+          const tableRes = await fetch(
+            `/api/w/${owner.sId}/data_sources/${selectedDataSource.name}/tables/${id}`
+          );
+          const { table } = (await tableRes.json()) as { table: CoreAPITable };
+          return table;
+        })
+      );
+      const configs = tables.map((table) => ({
+        workspaceId: owner.sId,
+        dataSourceId: table.data_source_id,
+        tableId: table.table_id,
+        tableName: table.name,
+      }));
+
+      onSave(configs);
+    } finally {
+      setIsSaving(false);
+    }
+  }
 
   const onClose = () => {
     setOpen(false);
     setTimeout(() => {
       setSelectedDataSource(null);
-      setSelectedTable(null);
+      setSelectedManagedTables([]);
+      setIsSaving(false);
+      setIsInitializingNodes(false);
+      setNodeIntializationError(false);
     }, 200);
   };
 
   return (
     <Modal
       isOpen={isOpen}
-      onClose={() => {
-        if (selectedDataSource !== null) {
-          setSelectedDataSource(null);
```

Comment:
This was needed to have a coherent navigation flow. Is it still the case with your new implementation?

Reply:
This is done in the onClose function.